traits/trait_types.py

```diff
@@ -2925,7 +2929,12 @@ def __init__(self, map, **metadata):
         try:
             default_value = metadata.pop("default_value")
         except KeyError:
-            default_value = next(iter(self.map))
+            if len(self.map) > 0:
+                default_value = next(iter(self.map))
+            else:
+                raise ValueError(
+                    "The dictionary of valid values can not be empty."
+                )
```

Comment:
You may want to make this a `raise ... from None`, so that we the `KeyError` doesn't show up in the traceback.